File: problems/scaler/hashing/count_distinct_elements.js
/**
 * Q. Count distinct elements
 *
 * Given an array A of N integers, return the number of unique elements in the array.
 */

function soln(A) {
  const hs = new Set();

  for (const i of A) {
    hs.add(i);
  }

  return hs.size;
}

console.log(soln([3, 4, 3, 6, 6]));

File: src/main/java/com/dsa/problems/scaler/hashing/count_distinct_elements.java
package com.dsa.problems.scaler.hashing;

import java.util.*;

public class count_distinct_elements {
  /**
   * Q. Count distinct elements
   *
   * Given an array A of N integers, return the number of unique elements in the array.
   */
  private static int countDistinctElements (ArrayList<Integer> A) {
    HashSet<Integer> hs = new HashSet<>();

    for(int i = 0; i < A.size(); i++) {
      hs.add(A.get(i));
    }

    return hs.size();
  }

  public static void main(String[] args) {
    System.out.println(countDistinctElements(new ArrayList<>(Arrays.asList(1,2,3,1,2,4,2))));
  }
}
